File: app/src/main/java/com/dougkeen/util/WakeLocker.java
package com.dougkeen.util;

import android.content.Context;
import android.os.PowerManager;

public abstract class WakeLocker {
    private static PowerManager.WakeLock wakeLock;

    public static void acquire(Context ctx) {
        release();

        PowerManager pm = (PowerManager) ctx
                .getSystemService(Context.POWER_SERVICE);
        wakeLock = pm.newWakeLock(PowerManager.PARTIAL_WAKE_LOCK,
                "BartRunner:Alarm");
        wakeLock.acquire(60_000L);
    }

    public static void release() {
        if (wakeLock != null && wakeLock.isHeld())
            wakeLock.release();
        wakeLock = null;
    }
}

File: app/src/main/kotlin/com/dougkeen/util/WakeLocker.kt
package com.dougkeen.util

import android.content.Context
import android.os.PowerManager

object WakeLocker {
    private var wakeLock: PowerManager.WakeLock? = null

    @JvmStatic
    fun acquire(context: Context) {
        release()
        val powerManager = context.getSystemService(Context.POWER_SERVICE) as PowerManager
        wakeLock = powerManager.newWakeLock(
            PowerManager.PARTIAL_WAKE_LOCK,
            "BartRunner:Alarm",
        ).also { it.acquire(60_000L) }
    }

    @JvmStatic
    fun release() {
        wakeLock?.takeIf { it.isHeld }?.release()
        wakeLock = null
    }
}
